setup.py

```diff
@@ -0,0 +1,192 @@
+    """
+    Custom setuptools extension that configures a CMake project.
+    """
+
+    def __init__(self,
+                 name: str,
+                 cmake_configure_options: List[str] = (),
+                 source_dir: str = str(Path(".").absolute()),
+                 cmake_build_type: str = "Release"):
+
+        super().__init__(name=name, sources=[])
+
+        if not Path(source_dir).absolute().is_dir():
+            raise ValueError(f"Directory '{source_dir}' does not exist")
+
+        self.cmake_build_type = cmake_build_type
+        self.source_dir = str(Path(source_dir).absolute())
+        self.cmake_configure_options = cmake_configure_options
+
+
+class BuildExtension(build_ext):
+    """
+    Setuptools build extension handler.
+    It processes all the extensions listed in the 'ext_modules' entry.
+    """
+
+    def run(self) -> None:
+        """
+        Process all the registered extensions executing only the CMakeExtension objects.
+        """
+
+        # Filter the CMakeExtension objects
+        cmake_extensions = [e for e in self.extensions if isinstance(e, CMakeExtension)]
+
+        if len(cmake_extensions) == 0:
+            raise ValueError("No CMakeExtension objects found")
+
+        # Check that CMake is installed
+        self._raise_if_not_installed(command="cmake")
+
+        # Check that Ninja is installed
+        self._raise_if_not_installed(command="ninja")
+
+        if platform.system() != "Linux":
+            raise RuntimeError("Only Linux is currently supported")
+
+        for ext in cmake_extensions:
+            self.build_extension(ext)
+
+    @staticmethod
+    def _raise_if_not_installed(command: str) -> None:
+
+        try:
+            _ = subprocess.check_output(['which', command])
+        except subprocess.CalledProcessError:
+            raise RuntimeError(f"Required command '{command}' not found")
+
+    def build_extension(self, ext: CMakeExtension) -> None:
+        """
+        Build a CMakeExtension object.
+
+        Args:
+            ext: The CMakeExtension object to build.
+        """
+
+        # Disable editable installation (pip install -e .)
+        if self.inplace:
+            raise RuntimeError("Editable installation is not supported")
+
+        # In normal installation, the ext_dir is the folder that is then compressed to
+        # make the archive. Once installed, it represents the site-package directory.
+        # Instead, in editable installation, the ext_dir is the directory that contains
+        # the in-source Python package. In this case, the CMake project would be installed
+        # in-source, and this is the reason why the editable installation is disabled.
+        ext_dir = Path(self.get_ext_fullpath(ext.name)).parent.absolute()
+        cmake_install_prefix = ext_dir / "idyntree"
+
+        # CMake configure arguments
+        configure_args = [
+            f"-GNinja",
+            f"-DCMAKE_INSTALL_PREFIX:PATH={cmake_install_prefix}",
```

Comment:
```suggestion
            f"-DCMAKE_INSTALL_PREFIX:PATH={cmake_install_prefix}",
            f"-DIDYNTREE_USES_IPOPT:BOOL=ON",
            f"-DIDYNTREE_USES_ASSIMP:BOOL=ON",
            f"-DIDYNTREE_USES_IRRLICHT:BOOL=OFF",
            f"-DIDYNTREE_USES_QT5:BOOL=OFF",
            f"-DIDYNTREE_USES_OSQPEIGEN:BOOL=OFF",
            f"-DIDYNTREE_USES_ALGLIB:BOOL=OFF",
            f"-DIDYNTREE_USES_WORHP :BOOL=OFF",
            f"-DIDYNTREE_USES_YARP:BOOL=OFF",
            f"-DIDYNTREE_USES_ICUB_MAIN:BOOL=OFF",
```

To ensure a reproducible package being generated regardless of the state of the dependency that are installled in the system, I would add here explicitly which dependency we build and which we don't . See https://github.com/robotology/idyntree/blob/devel/cmake/iDynTreeDependencies.cmake for more details.

Reply:
Good point! I update the PR accordingly.

Reply:
Done in 7b89565. The location where the CMake options are passed is not there but below, where the extension is configured. The lines of your suggestion are meant to be project-independent and could be moved to a standalone library.